demo/full/bxf-parser2/src/parser/schedule/index.js

```diff
@@ -0,0 +1,95 @@
+import reduceChildren from "./../reduce_children";
+import {
+  parseEventData,
+  parseContentData,
+} from "./schedule_elements";
+
+/**
+ * Parse entire schedule
+ * @param {Element} node
+ */
+export default function parseSchedule(node) {
+  return reduceChildren(node, (res, name, node) => {
+    switch (name.toLowerCase()) {
+      case "ns1:scheduledevent":
+        // const action = node.attributes[0].value; //action
+        res.push(parseScheduleEvent(node));
+    }
+    return res;
+  }, []);
+
+  /**
+   * Parse schedule event.
+   * A schedule event contains one or more schedule element.
+   * @param {Element} node
+   */
+  function parseScheduleEvent(node) {
+    return reduceChildren(node, (res, name, node) => {
+      switch (name.toLowerCase()) {
+        case "ns1:scheduleelements":
+          const {
+            eventData,
+            contentData,
+          } = parseScheduleElements(node);
+          if (eventData && contentData) {
+            if (eventData.video) {
+              if (res.video === undefined) {
+                res.video = [];
+              }
+              res.video.push({
+                title: eventData.video.title,
+                startTime: eventData.video.startTime,
+                endTime: eventData.video.startTime + eventData.video.duration,
+                affaire: contentData ? parseInt(contentData.prod) : undefined,
+                pgrm: contentData ? parseInt(contentData.pgrm) : undefined,
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Change this condition so that it does not always evaluate to "true"; some subsequent code is never executed. ([javascript:S2583](https://sonarcloud.io/organizations/rx-player/rules#rule_key=javascript%3AS2583))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=366&issues=AWbt7Z3qW3Y69bz5aGXB&open=AWbt7Z3qW3Y69bz5aGXB)

[ISSUE_KEY]: AWbt7Z3qW3Y69bz5aGXB